src/program_generator/pg_marginal.py

```diff
@@ -166,6 +170,42 @@ def _generate_call_program(create_operation, op_count, max_op_count):
   opcodes_with_pops = (create_operation['Value'][2:4] + '50') * op_count
   pops = '50' * (max_op_count - op_count + 5)
   return empty_pushes + account_deployment_code + '60ff' + opcode_args + opcodes_with_pops + pops
+
+def _generate_log_program(log_operation, op_count, max_op_count):
+  """
+  Generates a program for LOG* opcodes
+  """
+  assert log_operation['Mnemonic'] in ['LOG0', 'LOG1', 'LOG2', 'LOG3', 'LOG4']
+  
+  # fill out memory
+  memory_filler_push = '61ffff'
+  memory_address = 0
+  memory = ''
+  for i in range(100):
+    memory_address += 32
+    memory_address_hexed = hex(memory_address)[2:]
+    memory_address_hexed = '0' * (4 - len(memory_address_hexed)) + memory_address_hexed
+    memory += memory_filler_push + '61' + memory_address_hexed + '52'
+  
+  arguments = '60ff' * arity(log_operation) * max_op_count
+  logs = log_operation['Value'][2:4] * op_count
+
+  return memory + arguments + logs
+
+def _generate_revert_program(operation, op_count, max_op_count):
+  """
+  Generates a program for REVERT and RETURN opcodes
+  """
+  assert operation['Mnemonic'] in ['REVERT', 'RETURN']
+
+  op_deployment_code = '726960ff60005260026018f3600052600a6016' + operation['Value'][2:4] + '6000526013600d6000f0'
```

Comment:
Please double check.
REVERT/RETURN is put at the end of the constructor, not the new contract bytecode. Similar with `no_op_deployment_code`, it is missing RETURN in the constructor. Actually I am not sure what happens in that case, I bet the new contract is not deployed successfully.

Reply:
Right. Thanks. I also figured that out today when I wanted to do STOP opcode.